roles/ec2_instance_create/tasks/ec2_instance_create_operations.yml

```diff
@@ -0,0 +1,101 @@
+---
+- name: Verify exclusive parameters
+  block:
+    - name: Check for security group ID
+      ansible.builtin.fail:
+        msg: "ec2_instance_create_external_sg_id and ec2_instance_create_associate_external_sg are mutually exlcusive.
+              Please provide only one to either associate existing or create new sg."
+      when: ec2_instance_create_external_sg_id is defined and ec2_instance_create_associate_external_sg is defined and ec2_instance_create_external_sg_id != None and ec2_instance_create_associate_external_sg is true
+
+- name: Verify that the instance and security group with same name does not exist
+  block:
+    - name: Get instane info with provided name
+      amazon.aws.ec2_instance_info:
+        region: "{{ ec2_instance_create_aws_region }}"
+        filters:
+          tag:Name: "{{ ec2_instance_create_instance_name }}"
+          instance-state-name: ["running"]
+      register: ec2_info_result
+
+    - name: Print warning and exit
+      ansible.builtin.fail:
+        msg: "Instance with name {{ ec2_instance_create_instance_name }} already exists in {{ ec2_instance_create_aws_region }}.
+              Please provide different name to avoid updating instance."
+      when: ec2_info_result.instances | length >= 1
+
+- name: Create EC2 instance with provided configuration
+  amazon.aws.ec2_instance:
+    region: "{{ ec2_instance_create_aws_region }}"
+    name: "{{ ec2_instance_create_instance_name }}"
+    instance_type: "{{ ec2_instance_create_instance_type }}"
+    image_id: "{{ ec2_instance_create_ami_id }}"
+    key_name: "{{ ec2_instance_create_key_name }}"
+    vpc_subnet_id: "{{ ec2_instance_create_vpc_subnet_id }}"
+    security_group: "{{ ec2_instance_create_external_sg_id | default(omit) }}"
+    tags: "{{ ec2_instance_create_tags | default(omit) }}"
+    wait: "{{ ec2_instance_create_wait_for_boot }}"
+  register: ec2_instance
+
+- name: Create security group if enabled
+  when: ec2_instance_create_associate_external_sg is true
+  block:
+    - name: Define security group
+      amazon.aws.ec2_security_group:
+        name: "{{ ec2_instance_create_external_sg_name | default('ec2_instance_create-default-external-sg') }}"
```

Comment:
Are these two defaults needed? It looks like they're already defined in the role defaults.